client-spark/common/src/main/java/org/apache/spark/shuffle/reader/RssShuffleDataIterator.java

```diff
@@ -106,8 +107,17 @@ public boolean hasNext() {
       shuffleReadMetrics.incFetchWaitTime(fetchDuration);
       if (compressedData != null) {
         shuffleReadMetrics.incRemoteBytesRead(compressedData.limit() - compressedData.position());
+        // Directbytebuffers are not collected in time will cause executor easy 
+        // be killed by cluster managers(such as YARN) for using too much offheap memory
+        if (uncompressedData != null && uncompressedData.isDirect()) {
+          try {
+            RssShuffleUtils.destroyDirectByteBuffer(uncompressedData);
+          } catch (Exception e) {
+            throw new RuntimeException("Destroy DirectByteBuffer failed!", e);
```

Comment:
RuntimeException -> RssException
For Spark Client, we usually throw a RssException. So we can count how many fail tasks are due to rss by the type of exception.